### PR TITLE
New API basic support

### PR DESCRIPTION
**Pull request by pablossp**
## Short Description:

- Changed the api.activate_signature functions for the new api.set_signature and api.set_hash_lib functions.
- Modified the requirements.txt to include the latest pgoapi API.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 numpy==1.11.0
 networkx==1.11
--e git+https://github.com/pogodevorg/pgoapi.git/@3a02e7416f6924b1bbcbcdde60c10bd247ba8e11#egg=pgoapi
+-e git+https://github.com/pogodevorg/pgoapi.git#egg=pgoapi
```
Comment (mmixx) [inline requirements.txt]: I think you need the add the tag? To get the latest update from a branch? This is what I put on my `requirements.txt`:

```
-e git+https://github.com/pogodevorg/pgoapi.git/@v0.43.3-alpha#egg=pgoapi
```

Comment (WayneUong) [inline requirements.txt]: You can use
`-e git+https://github.com/pogodevorg/pgoapi.git/@1a112879f424b518f994b99f43859dd95cd43fdf#egg=pgoapi
`
for consistency

File: pokemongo_bot/__init__.py [hunk 2]
```diff
@@ -999,6 +1001,32 @@ def get_encryption_lib(self):
...
+    #I'm missing OSX libraries support, anyone that can help me on this.
+    def get_hash_lib(self):
+        if _platform == "Windows" or _platform == "win32":
+            # Check if we are on 32 or 64 bit
+            if sys.maxsize > 2**32:
+                file_name = 'niantichash64.dll'
+            else:
+                file_name = 'niantichash32.dll'
+        else:
+            file_name = 'libniantichash-linux-x86-64.so'
```
Comment (WayneUong) [inline pokemongo_bot/__init__.py]: will this work on mac osx?

Comment (WayneUong) [inline pokemongo_bot/__init__.py]: Also linux 32?

Comment (pablossp) [inline pokemongo_bot/__init__.py]: No I don't think so, I haven't worked on mac ever so I don't know what to check the _platform variable against? care to help me on that?

Comment (WayneUong) [inline pokemongo_bot/__init__.py]: it's 'darwin' on mine, not sure about the linux though

File: requirements.txt (hunk 1 above)
Comment (pablossp) [inline requirements.txt]: Thank you guys, I'll make the change, I knew i had to be missing something in case the repository gets updated

**Comment by pablossp** [comment]
I made some changes now.
I still need a condition for the freebsd-64.so file.
And the web map is not showing Pokemon or the player in the map so far.
The bot client is working correctly so far.

**Comment by enginebai** [comment]
```python
if _platform.lower() == "linux" or _platform.lower() == "linux2":
    file_name = 'src/pgoapi/pgoapi/lib/libniantichash-linux-x86-64.so'
```

In `get_encrypt_lib()`, the `file_name` should be `libencrypt-linux-x86-64.so'.

File: pokemongo_bot/__init__.py (hunk 2 above)
Comment (lukasz138) [inline pokemongo_bot/__init__.py]: it's not working on linux 32-bit

**Comment by pablossp** [comment]
@lukasz138 try @enginebai solution, i've updated my branch (in a very ugly way) in order for it to work on linux now.

**Comment by lukasz138** [comment]
Exception: Could not find linux2 encryption library /root/pgoapi/PokemonGo-Bot/src/pgoapi/pgoapi/lib/libencrypt-linux-x86-32.so
2016-11-07 14:16:36,491 [sentry.errors] [ERROR] Sentry responded with an error: 'module' object has no attribute 'HTTPSHandler' (url: https://app.getsentry.com/api/90254/store/)
Traceback (most recent call last):
  File "/root/pgoapi/PokemonGo-Bot/local/lib/python2.7/site-packages/raven/transport/threaded.py", line 174, in send_sync
    super(ThreadedHTTPTransport, self).send(data, headers)
  File "/root/pgoapi/PokemonGo-Bot/local/lib/python2.7/site-packages/raven/transport/http.py", line 47, in send
    ca_certs=self.ca_certs,
  File "/root/pgoapi/PokemonGo-Bot/local/lib/python2.7/site-packages/raven/utils/http.py", line 44, in urlopen
    class ValidHTTPSHandler(urllib2.HTTPSHandler):
AttributeError: 'module' object has no attribute 'HTTPSHandler'

**Comment by pablossp** [comment]
@lukasz138 run pip install -r requirements.txt to download the new API to the src directory.

**Comment by lukasz138** [comment]
still the same

**Comment by enginebai** [comment]
@lukasz138 I encountered this problem before, and solved by locating the current encryption and hash lib file. Make sure `./pokemon_bot/__init__.py` is updated as this PR.

**Comment by lukasz138** [comment]
Problem is in pgoapi/utilities.py : (there is no 32-bit libraries)

 elif sys.platform.startswith('linux'):
        if "centos" in platform.platform():
            if platform.architecture()[0] == '64bit':
                encrypt_lib = "libencrypt-centos-x86-64.so"
                hash_lib = "libniantichash-centos-x86-64.so"
            else:
                encrypt_lib = "libencrypt-linux-x86-32.so"
                hash_lib = "libniantichash-linux-x86-32.so"
        else:
            if platform.architecture()[0] == '64bit':
                encrypt_lib = "libencrypt-linux-x86-64.so"
                hash_lib = "libniantichash-linux-x86-64.so"
            else:
                encrypt_lib = "libencrypt-linux-x86-32.so"
                hash_lib = "libniantichash-linux-x86-32.so"

**Comment by nullpixel1** [comment]
Can you make the PR for extra libs here? https://github.com/pogodevorg/pgoapi/pull/

**Comment by lukasz138** [comment]
OK, tell me how ?

**Comment by nullpixel1** [comment]
This was intended for original op

**Comment by pablossp** [comment]
@nullpixel1 This same PR? to the pogo dev branch? Id be honored.
Please tell me exactly what you need and ill be glad to help.

**Comment by nullpixel1** [comment]
@pablossp Just PR your binaries 😀 We need people to compile them for us, as we don't have all the binaries

**Comment by pablossp** [comment]
Ohh, I think you might be mistaken, we are actually missing them, @lukasz138 is saying he is having an error because he does not have them. If I get to compile any of them when Im back home Ill be glad to send any missing binaries to the project.

**Comment by nullpixel1** [comment]
We've got a PR that's got them. I'll review and merge them later

**Comment by solderzzc** [comment]
👍

**Comment by solderzzc** [comment]
Let me merge and have it run on Linux :) Everybody can't wait to see the green log.

**Comment by nachi9211** [comment]
Is bot working after latest update with CP changes?

**Pull request merged**